Add const interators in PointCloud to cbegin cend

```
apps/in_hand_scanner/src/icp.cpp:407:33: error: no member named 'cbegin' in 'pcl::PointCloud<pcl::PointXYZRGBNormal>'; did you mean 'begin'?
  for (auto it_in = cloud_data->cbegin (); it_in!=cloud_data->cend (); ++it_in)
```

And pointcloud.h:
```
      inline iterator begin () { return (points.begin ()); }
      inline iterator end ()   { return (points.end ()); }
      inline const_iterator begin () const { return (points.begin ()); }
      inline const_iterator end () const  { return (points.end ()); }
```
`cloud_data` is const, so he should use correct version.

_Originally posted by @SunBlack in https://github.com/_render_node/MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDI1MjIwMDE1OA==/comments/review_comment_

--- a/common/include/pcl/point_cloud.h
+++ b/common/include/pcl/point_cloud.h
@@ -443,14 +443,14 @@ namespace pcl
       using const_reverse_iterator = typename VectorType::const_reverse_iterator;
       inline iterator begin () noexcept { return (points.begin ()); }
       inline iterator end () noexcept { return (points.end ()); }
-      inline iterator begin () const noexcept { return (points.begin ()); }
-      inline iterator end () const noexcept { return (points.end ()); }
+      inline const_iterator begin () const noexcept { return (points.begin ()); }
+      inline const_iterator end () const noexcept { return (points.end ()); }
       inline const_iterator cbegin () const noexcept { return (points.cbegin ()); }
       inline const_iterator cend () const noexcept { return (points.cend ()); }
       inline reverse_iterator rbegin () noexcept { return (points.rbegin ()); }
       inline reverse_iterator rend () noexcept { return (points.rend ()); }
-      inline reverse_iterator rbegin () const noexcept { return (points.rbegin ()); }
-      inline reverse_iterator rend () const noexcept { return (points.rend ()); }
+      inline const_reverse_iterator rbegin () const noexcept { return (points.rbegin ()); }
+      inline const_reverse_iterator rend () const noexcept { return (points.rend ()); }
       inline const_reverse_iterator crbegin () const noexcept { return (points.crbegin ()); }
       inline const_reverse_iterator crend () const noexcept { return (points.crend ()); }
 

--- a/common/include/pcl/point_cloud.h
+++ b/common/include/pcl/point_cloud.h
@@ -439,10 +439,20 @@ namespace pcl
       // iterators
       using iterator = typename VectorType::iterator;
       using const_iterator = typename VectorType::const_iterator;
-      inline iterator begin () { return (points.begin ()); }
-      inline iterator end ()   { return (points.end ()); }
-      inline const_iterator begin () const { return (points.begin ()); }
-      inline const_iterator end () const  { return (points.end ()); }
+      using reverse_iterator = typename VectorType::reverse_iterator;
+      using const_reverse_iterator = typename VectorType::const_reverse_iterator;
+      inline iterator begin () noexcept { return (points.begin ()); }
+      inline iterator end () noexcept { return (points.end ()); }
+      inline iterator begin () const noexcept { return (points.begin ()); }
+      inline iterator end () const noexcept { return (points.end ()); }
+      inline const_iterator cbegin () const noexcept { return (points.cbegin ()); }
+      inline const_iterator cend () const noexcept { return (points.cend ()); }
+      inline reverse_iterator rbegin () noexcept { return (points.rbegin ()); }
+      inline reverse_iterator rend () noexcept { return (points.rend ()); }
+      inline reverse_iterator rbegin () const noexcept { return (points.rbegin ()); }
+      inline reverse_iterator rend () const noexcept { return (points.rend ()); }
+      inline const_reverse_iterator crbegin () const noexcept { return (points.crbegin ()); }
+      inline const_reverse_iterator crend () const noexcept { return (points.crend ()); }
 
       //capacity
       inline std::size_t size () const { return (points.size ()); }

